src/sentry/static/sentry/app/views/organizationDiscover/sidebar/orderby.jsx

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import {Flex} from 'grid-emotion';
+
+import {t} from 'app/locale';
+import SelectControl from 'app/components/forms/selectControl';
+import {SidebarLabel} from '../styles';
+
+export default class Orderby extends React.Component {
+  static propTypes = {
+    value: PropTypes.string.isRequired,
+    columns: PropTypes.array.isRequired,
+    onChange: PropTypes.func.isRequired,
+    disabled: PropTypes.bool,
+  };
+
+  updateField(field) {
+    const orderby = this.getInternal(this.props.value);
+    orderby.field = field;
+    this.props.onChange(this.getExternal(orderby));
+  }
+
+  updateDirection(direction) {
+    const orderby = this.getInternal(this.props.value);
+    orderby.direction = direction;
+    this.props.onChange(this.getExternal(orderby));
+  }
+
+  /**
+   * @param {Object} value Object containing orderby information
+   * @returns {String} value
+   */
+  getExternal(value) {
+    return `${value.direction === 'desc' ? '-' : ''}${value.field}`;
+  }
+
+  /**
+   * @param {String} value String containing orderby information
+   * @returns {Object} value
+   */
+  getInternal(value) {
+    const direction = value.startsWith('-') ? 'desc' : 'asc';
+    const field = value.replace(/^-/, '');
+    return {
+      direction,
+      field,
+    };
+  }
+
+  render() {
+    const {disabled, columns, value} = this.props;
+
+    const {direction, field} = this.getInternal(value);
+
+    return (
+      <React.Fragment>
+        <SidebarLabel className="control-label">
+          {t('Order by')}
+        </SidebarLabel>
+        <Flex>
+          <OrderByField>
```

Comment:
nit: Can we make this `OrderbyField` instead of `OrderByField`? Same for the value component below